pkg/operation/botanist/component/kubeapiserverexposure/kube_apiserver_service.go

```diff
@@ -49,25 +49,30 @@ var (
 
 // ServiceValues configure the kube-apiserver service.
 type ServiceValues struct {
+	// AnnotationsFunc is a function that returns annotations that should be added to the service.
 	AnnotationsFunc func() map[string]string
-	SNIPhase        component.Phase
+	// IsShootService declares the service as being used for shoot clusters.
+	IsShootService bool
```

Comment:
Instead of configuring this explicitly here, can't we check it implicitly by checking whether the service's namespace starts with https://github.com/gardener/gardener/blob/3151ac7a5c086a9ee7a8707bf6cc33f4ba0a5988/pkg/apis/core/v1beta1/constants/types_constants.go#L797-L799?

Reply:
Yes, good idea. I will change it.

Reply:
Why would it be a problem? The approach with checking for `TechnicalIDPrefix` also works well for namespaces with the "old scheme"?

Reply:
My bad, I expected the technical prefix to be `shoot--`. Thanks 👍